### DISCUSSION TITLE
Capturing value of selected ui.command() within a ui.menu() component in a form card

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#2109] [Q&A] [answered]
I have a visualization in a form card, who's data needs to be toggled. My code is as follows:

```
q.page['most_graph'] = ui.form_card(
            box=ui.box(zone='top_g', order=1, width='50%'),
            title='',
            items=[

                ui.inline(
                    items=[
                        ui.text_m(content='Most Popular Symbols'),
                        ui.menu(items=[
                            ui.command(name='tog_pop_o', label='Overall', value='pop_o', ), #02-08-23
                            ui.command(name='tog_pop_3', label='Last 3 Months', value='pop_3'),
                            ui.command(name='tog_pop_today', label='Today', value='pop_today')]
                        )
                    ], align='start', justify='between'
                ),
                ui.visualization(name='tr_g',
                                 data=data('Symbol Value', 5, rows=data_df, pack=True),
                                 plot=ui.plot([
                                     ui.mark(coord='theta', type='interval', x='=Symbol', y='=Value', stack='auto',
                                             y_min=0, color='$amber')
                                 ]),
                                 events=['select_marks'], 
                                 ),
            ],
        )
```
I need to update the visulaization based on the selected value in ui.menu() i.e. IF the user selects 'Overall'  I need to display the graph with over all data, and If the user selects 'Last 3 months' I need to display the graph using data f last 3 months. I am trying to do this by setting the value of the ui.command() but I'm unable to do. Is there a better way to achieve this?
This is the code I'm using in my serve function to capture the value:
```
if q.args.tog_pop_o:
     q.page['most_graph'].tog_pop_o.value='pop_o'
```

**Comment by mturoci** [comment, marked as answer]
I am confused. You say you want to update the visualization data based on menu item clicked, but your code tries to update the command value, not the visualization.

Btw, commands cannot by accessed by name yet - https://github.com/h2oai/wave/discussions/2107.

**Reply by n-srinidhi** [reply]
Hi, I forgot to add that the data/visualization card is being updated by a background process. So, I'm setting the command value so as to access the value in the background process and update the visualization accordingly. The code for the background process is as follows:

```
async def polling_function(q: Q):
      if q.page['most_graph'].tog_pop_3.value=='pop_3':
          q.client.rec_dict = q.client.frame_dict['most_traded']['last_3_months'] 
      elif q.page['most_graph'].tog_pop_today.value=='pop_today':
           q.client.rec_dict = q.client.frame_dict['most_traded']['today']
      else:
           q.client.rec_dict = q.client.frame_dict['most_traded']['overall']
      await q.page.save()
```

**Reply by n-srinidhi** [reply]
@mturoci I was trying to access the `ui.command` by name, Once I used the older method using index, it worked. Thank you!

**Reply by mturoci** [reply]
Glad to hear. Note that `if q.page['most_graph'].tog_pop_3.value=='pop_3':` is not a guaranteed behavior. Wave cards are write-only. The reason this works for you is just a coincidence, but I wouldn't rely on it for future.

A better way is to store the state into `q.client.tog_pop_3` for example.

**Reply by n-srinidhi** [reply]
Understood. Thanks a lot!
